File: Python/Stack/ReverseStack.py
class Stack:
    def __init__(self,quantity):
        self.quantity = quantity
        self.top = -1
        self.stackArray = [0] * quantity

    def push(self,value):
        if len(self.stackArray) < self.top:
            return print("Stack is full.")
        self.top += 1;
        self.stackArray[self.top] = value

    def pop(self):
        if self.top >= 0:
            topValue = self.stackArray[self.top] 
            self.stackArray[self.top] = 0
            self.top -= 1
            return topValue
        else:
             print("No value in the stack to pop.")


    def isEmpty(self):
        return self.top == -1
    
    
    def insertAtBottom(self,value):
        if self.isEmpty():
            self.push(value)
        else:
            temp = self.pop()
            self.insertAtBottom(value)
            self.push(temp)
    
    def reverseStack(self):
        if self.isEmpty() == False:
            temp = self.pop()
            self.reverseStack()
            self.insertAtBottom(temp)


    def showStackArray(self):
        print(self.stackArray)


class ReverseStack(Stack):
    def __init__(self):
        pass
    def reverseStack(self,stack):
        if stack.isEmpty():
            print("No value in the stack to reverse")
            return
        stack.reverseStack()


stack = Stack(10)
rv = ReverseStack()
stack.push(10)
stack.push(20)
stack.push(30)
stack.push(40)
stack.push(50)
stack.showStackArray()
rv.reverseStack(stack)
stack.showStackArray()

File: Java/Stack/ReverseStack.java

class Stack {
    private int[] stackArray;
    private int top;

    public Stack(int capacity) {
        stackArray = new int[capacity];
        top = -1;
    }

    public void push(int number) {
        if (top < stackArray.length - 1) {
            stackArray[++top] = number;
        } else {
            System.out.println("The stack is full.");
        }
    }

    public int pop() {
        if (!isEmpty()) {
            return stackArray[top--];
        } else {
            System.out.println("No value is in the stack.");
            return -1;
        }
    }

    public void insertAtBottom(int value) {
        if (isEmpty()) {
            push(value);
        } else {
            int temp = pop();
            insertAtBottom(value);
            push(temp);
        }
    }

    public void reverse() {
        if (!isEmpty()) {
            int temp = pop();
            reverse();
            insertAtBottom(temp);
        }
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public void display() {
        if (isEmpty()) {
            System.out.println("The stack is empty.");
            return;
        }
        for (int i = 0; i <= top; i++) {
            System.out.print(stackArray[i] + " ");
        }
        System.out.println();
    }
}

public class ReverseStack {

    static void reverseStack(Stack stack) {
        if (stack.isEmpty()) {
            System.out.println("Failed to reverse stack. Stack is empty.");
            return;
        }
        stack.reverse();
    }

    public static void main(String[] args) {
        Stack myStack = new Stack(5);
        myStack.push(10);
        myStack.push(20);
        myStack.push(30);

        System.out.println("Original Stack:");
        myStack.display();

        reverseStack(myStack);

        System.out.println("Reversed Stack:");
        myStack.display();
    }
}
